sql-cli/sql_cli/workflow_directory_parser.py

```diff
@@ -0,0 +1,231 @@
+        target_full_directory.mkdir(parents=True, exist_ok=True)
+
+        target_path = target_full_directory / self.path.name
+
+        target_path.write_text(self.content)
+
+        return target_path.relative_to(self.target_directory)
+
+    def write_raw_content_to_target_path(self) -> None:
+        """
+        Writes both content and headers to the target directory.
+        This is because with the "render" function, we will still need
+        the headers for creating proper TransformOperators
+        """
+        target_full_directory = (
+            self.target_directory
+            / GENERATED_WORKFLOW_INCLUDE_DIRECTORY
+            / self.root_directory.name
+            / "/".join(self.get_sub_directories())
+        )
+        target_full_directory.mkdir(parents=True, exist_ok=True)
+
+        target_path = target_full_directory / self.path.name
+
+        target_path.write_text(self.raw_content)
+
+    @abstractmethod
+    def to_operator(self) -> Any:
+        raise NotImplementedError()
+
+
+class SqlFile(WorkflowFile):
+    """A SqlFile is equivalent to a transform step in the Astro SDK."""
+
+    def __init__(self, root_directory: Path, path: Path, target_directory: Path) -> None:
+        super().__init__(root_directory, path, target_directory, SQL_FILE_TYPE)
+
+    def to_operator(self) -> TransformOperator:
+        """
+        Converts SQLFile into a TransformOperator that can be added to a DAG.
+        Any relevant metadata from the file frontmatter will be passed to the TransformOperator,
+        though we do not pass parameter dependencies at this stage.
+
+        :return: a TransformOperator
+        """
+        kwargs = {
+            "conn_id": self.metadata.get("conn_id"),
+            "parameters": None,
+            "handler": None,
+            "database": self.metadata.get("database"),
+            "schema": self.metadata.get("schema"),
+            "python_callable": lambda: (str(self.path), None),
+            "sql": self.content,
+        }
+        if airflow.__version__.startswith("2.2."):
+            kwargs["op_args"] = []
+        return TransformOperator(**kwargs)
+
+
+class YamlFile(WorkflowFile):
+    operator_instance_builder_callable_map = {LOAD_FILE_OPERATOR: get_load_file_instance}
+
+    def __init__(self, root_directory: Path, path: Path, target_directory: Path) -> None:
+        super().__init__(root_directory, path, target_directory, YAML_FILE_TYPE)
+        with open(path) as yaml_file:
+            self.yaml_content = yaml.load(yaml_file, Loader=yaml.Loader)
+        self.operator = self._get_operator()
+        self.yaml_file_name = self.path.stem
+
+    def _get_operator(self) -> str:
+        operator = list(self.yaml_content.keys())[0]
+        if operator not in YamlFile.operator_instance_builder_callable_map:
+            raise NotImplementedError(f"Operator support for {operator} not available")
+        return operator
+
+    def to_operator(self) -> LoadFileOperator:
```

Comment:
```suggestion
    def to_operator(self) -> BaseOperator:
```
maybe? Or we manually extend e.g. `LoadFileOperator | ExportFileOperator | ...`.

WDYT?